resources/testharness.js

```diff
@@ -587,11 +587,12 @@ policies and contribution forms [3].
 
     function async_test(func, name, properties)
     {
-        if (tests.promise_setup_called) {
-            tests.status.status = tests.status.ERROR;
-            tests.status.message = '`async_test` invoked after `promise_setup`';
-            tests.complete();
+        // async_test are deferred when a promise_setup is pending.
```

Comment:
If we do this for async_test, we might want to do it for `test` too. (Or we might decide to only do it for `async_promise_test` and then we'll need to duplicate code instead of re-using `async_test`.)

Reply:
What do you prefer?